direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java

```diff
@@ -648,14 +649,24 @@ public void waitUntilReady() throws InterruptedException {
     if (position == Position.OLDEST) {
       // seek all partitions to oldest data
       if (offsets == null) {
+        boolean emptyPoll = true;
         if (consumer.assignment().isEmpty()) {
           // If we don't find assignment within timeout, poll results in IllegalStateException.
           // https://cwiki.apache.org/confluence/display/KAFKA/KIP-266%3A+Fix+consumer+indefinite+blocking+behavior
-          consumer.poll(Duration.ofMillis(accessor.getAssignmentTimeoutMillis()));
+          emptyPoll =
+              consumer.poll(Duration.ofMillis(accessor.getAssignmentTimeoutMillis())).isEmpty();
+        }
+        final Set<TopicPartition> assignment = consumer.assignment();
+        final Map<TopicPartition, OffsetAndMetadata> committed = consumer.committed(assignment);
+        if (committed.values().stream().noneMatch(Objects::nonNull)) {
+          log.info("Seeking consumer name {} to beginning of partitions {}", name, assignment);
+          consumer.seekToBeginning(assignment);
+        } else {
+          if (!emptyPoll) {
+            log.info("Seeking consumer name {} to committed offsets {}", name, committed);
```

Comment:
This branch looks untested, can we fix that?

Reply:
this is cover by Kafka IT